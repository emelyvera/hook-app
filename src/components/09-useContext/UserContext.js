import { createContext } from 'react';

// este es un hired order component, eso es que tene coimponentes hijos 
export const UserContext = createContext( null );