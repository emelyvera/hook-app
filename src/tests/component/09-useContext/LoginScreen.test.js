import React from "react";
import { mount } from 'enzyme';
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en el componente <LoginScreen />', () => {

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />

        </UserContext.Provider>
    );

    test('should de mostarrse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('should de ejecutar el setUser con el argumento esprado', () => {
/**
 * cualquiera de las siguientes dos, son lo mismo 
 */
        //wrapper.find('button').simulate('click');
        wrapper.find('button').prop('onClick')();


        expect(setUser).toHaveBeenCalledWith({
            id: 123,
            name: 'emely'
        })

    });

});
