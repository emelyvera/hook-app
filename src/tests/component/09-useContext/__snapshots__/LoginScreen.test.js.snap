// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginScreen /> should de mostarrse correctamente 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
